tests/integration/features/cleanup_int_test.go

```diff
@@ -69,18 +70,99 @@ var _ = Describe("feature cleanup", func() {
 			Expect(featuresHandler.Delete()).To(Succeed())
 
 			// then
-			Eventually(createdSecretHasOwnerReferenceToOwningFeature(namespace, secretName)).
+			Eventually(createdSecretHasOwnerReferenceToOwningFeature(namespace)).
 				WithTimeout(fixtures.Timeout).
 				WithPolling(fixtures.Interval).
 				Should(WithTransform(errors.IsNotFound, BeTrue()))
 		})
 
 	})
 
+	Context("cleaning up conditionally enabled features", Ordered, func() {
+
+		const (
+			featureName = "enabled-conditionally"
+			secretName  = "test-secret"
+		)
+
+		var (
+			dsci            *dsciv1.DSCInitialization
+			namespace       string
+			featuresHandler *feature.FeaturesHandler
+		)
+
+		BeforeAll(func() {
+			namespace = envtestutil.AppendRandomNameTo("test-conditional-cleanup")
+			dsci = fixtures.NewDSCInitialization(namespace)
+		})
+
+		It("should create feature, apply resource and create feature tracker", func() {
+			// given
+			err := fixtures.CreateOrUpdateNamespace(envTestClient, fixtures.NewNamespace("conditional-ns"))
+			Expect(err).To(Not(HaveOccurred()))
+
+			featuresHandler = feature.ClusterFeaturesHandler(dsci, func(handler *feature.FeaturesHandler) error {
+				conditionalCreationErr := feature.CreateFeature(featureName).
+					For(handler).
+					UsingConfig(envTest.Config).
+					PreConditions(
+						feature.CreateNamespaceIfNotExists(namespace),
+					).
+					EnabledWhen(namespaceExists).
+					WithResources(fixtures.CreateSecret(secretName, namespace)).
+					Load()
+
+				Expect(conditionalCreationErr).ToNot(HaveOccurred())
+
+				return nil
+			})
+
+			// when
+			Expect(featuresHandler.Apply()).Should(Succeed())
+
+			// then
+			Eventually(createdSecretHasOwnerReferenceToOwningFeature(namespace)).
+				WithTimeout(fixtures.Timeout).
+				WithPolling(fixtures.Interval).
+				Should(Succeed())
+		})
+
+		It("should clean up resources when the condition is no longer met", func() {
+			// given
+			err := envTestClient.Delete(context.Background(), fixtures.NewNamespace("conditional-ns"))
+			Expect(err).To(Not(HaveOccurred()))
+
+			// Mimic reconcile by re-loading the feature handler
+			featuresHandler = feature.ClusterFeaturesHandler(dsci, func(handler *feature.FeaturesHandler) error {
+				conditionalCreationErr := feature.CreateFeature(featureName).
+					For(handler).
+					UsingConfig(envTest.Config).
+					PreConditions(
+						feature.CreateNamespaceIfNotExists(namespace),
+					).
+					EnabledWhen(namespaceExists).
+					WithResources(fixtures.CreateSecret(secretName, namespace)).
+					Load()
+
+				Expect(conditionalCreationErr).ToNot(HaveOccurred())
+
+				return nil
+			})
+
+			Expect(featuresHandler.Apply()).Should(Succeed())
+
+			// then
+			Eventually(createdSecretHasOwnerReferenceToOwningFeature(namespace)).
```

Comment:
I wonder if we shouldn't test something different here. We should ensure nothing that is supposed to be created by the defined feature (including tracker) should exist in the cluster.

On top of that, `Consistently` might be better instead of `Eventually` (which can pass in the first run). The clean-up from the previous run might not be done yet on the cluster when this test is executed. I see you are handling it with the current implementation by checking if ns is in a deletion state, though I believe switching it to `Consistently` should make it safer if we happen to rework the test fixture, but keep the expectations the same.